### PR TITLE
Add GUI section for Getting Started

### PR DESCRIPTION
**Pull request by FlorianJacta**
For issue: #564

### CODE REVIEW

File: docs/getting_started/index.md [hunk 1]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+page = """
+Name: <|{name}|input|>
+<|submit|button|on_action=submit_scenario|>
+Message: <|{message}|text|>
+"""
+```
+
+- The `page` variable is a Markdown representation of the user interface. It uses standard Markdown syntax as well as visual elements.
+
+- A Taipy visual element is of the form of `<|{variable}|visual_element_type|...|>`. Taipy has a lot of visual elements to play with, which interact with your Python variables and environment. `{variable}` is the main property of the visual element and is usually what is displayed or modified through the visual element.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- A Taipy visual element is of the form of `<|{variable}|visual_element_type|...|>`. Taipy offers various visual elements that can interact with your Python variables and environment. The main feature of these visual elements is typically the `{variable}`, which is what they display or change.
```

File: docs/getting_started/index.md [hunk 2]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+
+- For instance, `{name}` is bound to an input and text field, allowing the user’s input to be directly stored in the `name` variable.
+
+- Similarly, the `message` variable is bound to a text field, making it possible to display dynamic content to the user.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
- In the same way, the `message` variable is connected to a text field, allowing you to display changing content to the user.
```

File: docs/getting_started/index.md [hunk 3]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+- Actions, like `on_action=submit_scenario`, allow visual elements like buttons to trigger specific functions, enhancing the interactivity of the application.
+
+```python
+def submit_scenario(state):
+    scenario.name.write(state.name)
+    scenario.submit()
+    state.message = scenario.message.read()
+```
+
+Every callback, including `submit_scenario`, utilizes `state` as the first parameter. The `state` represents a user connection and is used to read and assign runtime variables for a specific user. It enables Taipy to be easily multi-user.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
In every callback, like `submit_scenario`, the first thing it uses is `state`. This state stands for a user's connection 
and helps read and set variables while the user is interacting with the program. 
This makes it simple for Taipy to handle multiple users at once.
```

File: docs/getting_started/index.md [hunk 4]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+
+While `scenario` remains globally accessible, `state.name` and `state.input` are specific to the user defining it. This architecture ensures that individual user interactions are isolated and managed efficiently while variables like `scenario` are accessed globally.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
While `scenario` is available to everyone using the program, `state.name` and `state.input` are unique to the user 
who defines them. This design ensures that each user's actions are separate and efficiently controlled, 
while variables like `scenario` are available to all users.
```

File: docs/getting_started/index.md [hunk 5]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+
+TODO: State illustration
+
+In the `submit_scenario` function, the `name` on the user interface is written to the scenario. Upon submission, the result is read into the `message` variable, which is then reflected in the UI thanks to the binding.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
In the `submit_scenario` function, the `name` entered by the user on the interface is saved to the scenario. 
After submission, the outcome is retrieved and stored in the `message` variable, which is then shown on the user interface due to the connection between them.
```

File: docs/getting_started/index.md [hunk 6]
```diff
@@ -117,35 +119,83 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+
+```python
+if __name__ == "__main__":
+    tp.Core().run()
+    scenario = tp.create_scenario(scenario_cfg)
+    Gui(page).run()
+```
 
-![GUI Result](result.png){width=50%}
+The application's main execution block initializes the Core services, creates a scenario, and launches the GUI, bringing the interactive interface to life.
```
Comment (erikfTW) [inline docs/getting_started/index.md]:
```suggestion
The main part of the application starts by setting up the Core services, generating a scenario, and starting the GUI, which makes the interactive interface active and functional.
```

**Review by erikfTW — CHANGES_REQUESTED** [review]

File: docs/getting_started/index.md (hunk 3 above)
Comment (FlorianJacta) [reply docs/getting_started/index.md]: I don't really like this formulation.

Comment (erikfTW) [reply docs/getting_started/index.md]: Then let's try: 

"Every callback, including `submit_scenario`, utilizes `state` as the first parameter. This `state` represents a user's connection and is used to read and set variables while the user is interacting with the program. It makes it simple for Taipy to handle multiple users simultaneously."

File: docs/getting_started/index.md [hunk 7]
```diff
@@ -49,14 +51,14 @@ The graph involves:
         return f"Hello {name}!"
     
     
-    name_data_node_cfg = Config.configure_data_node(id="input_name")
+    name_data_node_cfg = Config.configure_data_node(id="name")
```
Comment (jrobinAV) [inline docs/getting_started/index.md]: Nop name is a reserved attribute name. I would revert the change or find another name.

Please run the code to be sure.

File: docs/getting_started/index.md [hunk 8]
```diff
@@ -117,37 +119,103 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
 
 Message: <|{message}|text|>
 """
 
+name = "Taipy"
+message = None
+
+
+def submit_scenario(state):
+    state.scenario.name.write(state.name)
```
Comment (jrobinAV) [inline docs/getting_started/index.md]: This raises an error because scenario.name is not a data node but a string.
"name" is a reserved attribute name. you cannot name your data node like that.

File: docs/getting_started/index.md [hunk 9]
```diff
@@ -117,37 +119,103 @@ Here's a simple GUI set up for our "Hello World" scenario:
...
+
+Every callback, including `submit_scenario`, utilizes state as the first parameter. 
+This state represents a user's connection and is used to read and set variables while 
+the user is interacting with the program. It makes it simple for Taipy to handle multiple users simultaneously.
+
+While *scenario* is available to everyone using the program, *state.name* and *state.input* are unique to the user 
+who defines them. This design ensures that each user's actions are separate and efficiently controlled, 
+while variables like *scenario* are global variables.
+
+TODO: State illustration
```
Comment (jrobinAV) [inline docs/getting_started/index.md]: Do you have an illustration in mind?
Do you plan to add it in this PR or should we create a ticket for that?

**Review by jrobinAV — CHANGES_REQUESTED** [review]

File: docs/getting_started/index.md (hunk 9 above)
Comment (FlorianJacta) [reply docs/getting_started/index.md]: I asked Marine to work on it. I don't know if she would have the time to create one quickly so it will maybe be another PR

File: docs/getting_started/index.md (hunk 7 above)
Comment (FlorianJacta) [reply docs/getting_started/index.md]: Oh true, I put name as it is a bit simpler and also corresponds to the DAG of the Getting Started

File: docs/getting_started/index.md [hunk 10]
```diff
@@ -34,8 +34,10 @@ Let's craft a basic "Hello World" scenario:
 ![Hello World Example](hello_world.svg){width=50%}
 
 The graph involves:
-- An input data node named *name*.
-- A task, *build_message*, that processes the *name* data node and outputs a *message* data node.
+
+- An input data node named *input_name*.
+
+- A task, *build_message*, that processes the *input_name* data node and outputs a *message* data node.
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: Just wondering about the wording.
Do you want to equate tasks and functions?
(I mean: "A task, referring to the function\*build_message()\*, ...")

File: docs/getting_started/index.md [hunk 11]
```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
...
+page = """
+Name: <|{input_name}|input|>
+<|submit|button|on_action=submit_scenario|>
+Message: <|{message}|text|>
+"""
+```
+
+**Pages**
+
+Taipy pages can be defined in multiple ways: Markdown, Html or Python. The `page`
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: \`page\` -> \*page\*

File: docs/getting_started/index.md [hunk 12]
```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
...
+Actions, like `on_action=submit_scenario`, allow visual elements like buttons to trigger specific functions, enhancing the interactivity of the application.
+
+```python
+def submit_scenario(state):
+    scenario.input_name.write(state.input_name)
+    scenario.submit()
+    state.message = scenario.message.read()
+```
+
+Every callback, including `submit_scenario`, utilizes state as the first parameter.
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: \`submit_scenario\` -> \*submit_scenario()\*
"utilizes state as the first parameter." -> "receives a  \`State^\` object as its first parameter.'

File: docs/getting_started/index.md [hunk 13]
```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
...
+```python
+def submit_scenario(state):
+    scenario.input_name.write(state.input_name)
+    scenario.submit()
+    state.message = scenario.message.read()
+```
+
+Every callback, including `submit_scenario`, utilizes state as the first parameter. 
+This state represents a user's connection and is used to read and set variables while 
+the user is interacting with the program. It makes it simple for Taipy to handle multiple users simultaneously.
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: "It makes it simple" -> "It makes it possible"

File: docs/getting_started/index.md [hunk 14]
```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
...
+
+While *scenario* is available to everyone using the program, *state.input_name* and *state.input* are unique to the user
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: "unique to the user who defines them" -> "specific to the user who interacts with them"

File: docs/getting_started/index.md [hunk 15]
```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
...
+This state represents a user's connection and is used to read and set variables while 
+the user is interacting with the program. It makes it simple for Taipy to handle multiple users simultaneously.
+
+While *scenario* is available to everyone using the program, *state.input_name* and *state.input* are unique to the user 
+who defines them. This design ensures that each user's actions are separate and efficiently controlled, 
+while variables like *scenario* are global variables.
+
+TODO: State illustration
+
+In the `submit_scenario` function, the *input_name* entered by the user on the interface is saved to the scenario.
```
Comment (FabienLelaquais) [inline docs/getting_started/index.md]: \`submit_scenario\` -> \*submit_scenario()\*

**Review by FabienLelaquais — APPROVED** [review]
Typographic changes, please.

File: docs/getting_started/index.md (hunk 10 above)
Comment (FlorianJacta) [reply docs/getting_started/index.md]: It is quite the same for me. But I think with your formulation, it also defines what is a task directly

**Review by erikfTW — APPROVED** [review]

**Review by jrobinAV — APPROVED** [review]
